source/tank/controller/shoot/AutoAimController.hx

```diff
@@ -0,0 +1,46 @@
+package tank.controller.shoot;
+
+import flixel.FlxObject;
+import flixel.math.FlxMath;
+import flixel.math.FlxPoint;
+
+class AutoAimController extends BaseController implements IShootController {
+	static inline public var SHOOT_TIMER_MAX = .5;
+
+	private var shootTimerRemaining:Float = SHOOT_TIMER_MAX;
+
+	private var rotateBy:Float = 0;
+
+	private var targetTank:FlxObject;
+
+	public function new(controlledTank:Tank, target:FlxObject) {
+		super(controlledTank);
+		targetTank = target;
+	}
+
+	private function distanceToTarget():Float {
+		var dx = (controlledTank.x - targetTank.x);
+		var dy = (controlledTank.y - targetTank.y);
+		return Std.int(FlxMath.vectorLength(dx, dy));
+	}
+
+	override public function update(elapsed:Float):Void {
+		shootTimerRemaining -= elapsed;
+		rotateBy = controlledTank.getPosition().degreesTo(targetTank.getPosition());
+		if (shouldShoot()) {
+			// shoot bullet if shootShoot true
+		}
```

Comment:
The `Tank` class does this already, actually, so you don't have to worry about putting it in your aim controller:
https://github.com/SamBumgardner/csc-303-game-2023/blob/main/source/tank/Tank.hx#L94-L96

the controller just has to define a cool `shouldShoot()` method (like you have already) and `Tank` handles calling it and executing whatever necessary logic is required afterward.

If you want, you could remove this as part of this PR. We can also just come by and clean it up as we take care of bullet stuff.